### PR TITLE
feat(client)!: Return a friendly type from handle.describe()

### LINKED ISSUE #482: [Feature Request] Wrap `WorkflowHandle.describe` output in a JS friendly object
### Is your feature request related to a problem? Please describe.

image

image

image

https://github.com/protobufjs/protobuf.js/pull/1258

### PR DESCRIPTION
**Pull request by lorensr**
Closes #482

### CODE REVIEW

File: packages/client/src/types.ts [hunk 1]
```diff
@@ -10,3 +10,19 @@ export type DescribeWorkflowExecutionResponse = temporal.api.workflowservice.v1.
 export type TerminateWorkflowExecutionResponse = temporal.api.workflowservice.v1.ITerminateWorkflowExecutionResponse;
 export type RequestCancelWorkflowExecutionResponse =
   temporal.api.workflowservice.v1.IRequestCancelWorkflowExecutionResponse;
+
+export interface WorkflowExecutionDescription {
+  type: string;
+  workflowId: string;
+  runId?: string;
```
Comment (bergundy) [inline packages/client/src/types.ts:17]: I don't think this is optional

File: packages/client/src/types.ts [hunk 2]
```diff
@@ -10,3 +10,19 @@ export type DescribeWorkflowExecutionResponse = temporal.api.workflowservice.v1.
...
+  taskQueue: string;
+  status: temporal.api.enums.v1.WorkflowExecutionStatus;
+  historyLength: Long;
+  startTime: Date;
+  executionTime?: Date;
+  closeTime?: Date;
+  memo?: temporal.api.common.v1.IMemo;
```
Comment (bergundy) [inline packages/client/src/types.ts:24]: We should decode the payloads here

File: packages/client/src/types.ts [hunk 3]
```diff
@@ -10,3 +10,19 @@ export type DescribeWorkflowExecutionResponse = temporal.api.workflowservice.v1.
...
+  searchAttributes?: temporal.api.common.v1.ISearchAttributes;
```
Comment (bergundy) [inline packages/client/src/types.ts:25]: Same

File: packages/client/src/types.ts [hunk 4]
```diff
@@ -10,3 +10,19 @@ export type DescribeWorkflowExecutionResponse = temporal.api.workflowservice.v1.
...
+  parentExecution?: temporal.api.common.v1.IWorkflowExecution;
```
Comment (bergundy) [inline packages/client/src/types.ts:26]:
```suggestion
  parentExecution?: Required<temporal.api.common.v1.IWorkflowExecution>;
```

**Review by bergundy — APPROVED** [review]
Pre-approving

**Comment by lorensr** [comment]
`handle.describe()` returns:

image

**Pull request merged**